src/Service/AddBulletinToQueue.php

```diff
@@ -0,0 +1,602 @@
+<?php
+
+namespace Drupal\govdelivery_bulletins\Service;
+
+use Drupal\Component\Utility;
+use Drupal\Core\Queue\QueueFactory;
+use Drupal\Core\Queue\QueueInterface;
+
+/**
+ * Class for handling adding a GovDelivery bulletin to the queue.
+ */
+class AddBulletinToQueue {
+
+  // Queue Operation flags.
+  /**
+   * Flag to set whether the queue should be deduped as this bulletin is added.
+   *
+   * @var bool
+   */
+  private $flag_dedupe = FALSE;
+
+  /**
+   * Flag to do this bulletin as a GovDelivery Test.
+   *
+   * @var bool
+   */
+  private $flag_test = FALSE;
+
+  // Queue Operation variables.
+  /**
+   * An of status, warnings, and error mesages.
+   *
+   * @var array
+   */
+  private $messages = [
+    'error' => [],
+    'status' => [],
+    'warning' => [],
+  ];
+
+  /**
+   * An id for the queue to use in deduping the queue.
+   *
+   * If dedupe flag set, any other queue items with this id will be removed.
+   *
+   * @var string
+   */
+  private $queueUid = NULL;
```

Comment:
The naming of this was/is confusing to me. We can run with it though and see how it goes.